Add cluster API resource faq (#488)
karmada-io/website: docs/faq/faq.md

  conflicts.

TODO: Link to adoption use case once it gets on board.

## Why Cluster API doesn't have the CRD yaml file?

> Kubernetes provides two methods to extend APIs: Custom Resource and Kubernetes API Aggregation Layer, for more detail, we can refer to [Extending the Kubernetes API](https://kubernetes.io/docs/concepts/extend-kubernetes/).

Reviewer: ```suggestion
Kubernetes provides two methods to extend APIs: Custom Resource and the Kubernetes API Aggregation Layer. For more details, you can refer to [Extending the Kubernetes API](https://kubernetes.io/docs/concepts/extend-kubernetes/).
```
In general, when referring to the reader or user, it is common to use "you."
On the other hand, "we" typically represents the developers or authors.
Therefore, it is preferred to use "you" when explaining concepts or instructions to the reader.

Author: Thanks a lot~

> Karmada uses the two extension methods. For example, `PropagationPolicy` and `ResourceBinding` use the Custom Resource method, and `Cluster` resource use the `Kubernetes API Aggregation Layer` method.

Reviewer 2: It would be great to explain a little bit more details about why Cluster needs to be extended by aggregated layer.

Author: Ok, let me do it.

> Karmada uses the two extension methods. For example, `PropagationPolicy` and `ResourceBinding` use the Custom Resource method, and `Cluster` resource use the `Kubernetes API Aggregation Layer` method.

Reviewer: ```suggestion
Karmada uses the both extension methods. For example, `PropagationPolicy` and `ResourceBinding` use the Custom Resource, and `Cluster` resource uses the `Kubernetes API Aggregation Layer`.
```
Not very necessary to put `method` everywhere.

> Therefore, `Cluster` resources do not have CRD yaml file, and we can not see the `Cluster` when we execute `kubectl get crd` command.

Reviewer: ```suggestion
Therefore, `Cluster` resources do not have a CRD YAML file, and they are not visible when you execute the `kubectl get crd` command.
```

  conflicts.

TODO: Link to adoption use case once it gets on board.

## Why Cluster API doesn't have the CRD YAML file?

Kubernetes provides two methods to extend APIs: Custom Resource and Kubernetes API Aggregation Layer. For more detail, you can refer to [Extending the Kubernetes API](https://kubernetes.io/docs/concepts/extend-kubernetes/).

> Karmada uses the both extension methods. For example, `PropagationPolicy` and `ResourceBinding` use the `Custom Resource`, and `Cluster` resource uses the `Kubernetes API Aggregation Layer`.

Reviewer: ```suggestion
Karmada uses the both extension methods. For example, `PropagationPolicy` and `ResourceBinding` use Custom Resource, and `Cluster` resource uses Kubernetes API Aggregation Layer.
```

Therefore, `Cluster` resources do not have a CRD YAML file, and they are not visible when you execute the `kubectl get crd` command.

> So, why do we use the `Kubernetes API aggregation layer` to extend `Cluster` resources instead of using `custom resources`?

Reviewer: ```suggestion
So, why would you choose to use the Kubernetes API Aggregation Layer to extend `Cluster` resources instead of using Custom Resources?
```

> This is because we need to set up the `proxy` sub-resource for the `Cluster` resource. By using `proxy`, users can access resources in member clusters. For details, please refer to [Aggregation Layer APIServer](https://karmada.io/ en/docs/next/userguide/globalview/aggregated-api-endpoint). Currently, the `custom resource` does not support setting `proxy` sub-resources, which is why we did not choose it.

Reviewer: ```suggestion
This is because the `Cluster` resource requires the setup of the `proxy` sub-resource. By using `proxy`, you can access resources in member clusters. For more information, refer to [Aggregation Layer APIServer](https://karmada.io/docs/next/userguide/globalview/aggregated-api-endpoint/). At present, Custom Resources do not support configuring `proxy` sub-resources, which is why it was not chosen for this purpose.
```
